DAQ_FW/lib/IADCSensor/IADCSensor.h

```diff
@@ -0,0 +1,31 @@
+#ifndef IADCSENSOR_LIB
+#define IADCSENSOR_LIB
+
+#include "Logger.h"
+#include "system_config.h"
+
+class IADCSensor
+{
+private:
+    // bit to voltage conversion factor
+    float mVOLTAGE_PER_BIT;
+
+    // SENSOR METADATA:
+    // Currently Generic Parameters, to be expanded as needed
+    // TODO: add more specific data, depends on the sensors
+    const char *mSensorName;
+    uint32_t mSensorID;
+    uint32_t mADC_ID;
+
+    void ADCcovert(uint16_t _ADC_GAIN_MODE);
+
+public:
+    IADCSensor(const char *_SensorName, uint16_t _SensorID, uint16_t _ADC_ID, uint16_t _ADC_GAIN_MODE);
+
+    // Read function - to be overwritten by the Driver class
+    virtual void Read(){};
```

Comment:
I also think this method should be const. This one is even more important because it ensures that @lonamle doesn't accidentally modify the member variables you declared and gives her more info on how to write her function